core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/documents/UINewDocumentForm.java

```diff
@@ -83,10 +86,11 @@
    * Constructor.
    *
    */
-  public UINewDocumentForm() {
+  public UINewDocumentForm() throws Exception{
     this.documentService = this.getApplicationComponent(DocumentService.class);
     // Title textbox
     UIFormStringInput titleTextBox = new UIFormStringInput(FIELD_TITLE_TEXT_BOX, FIELD_TITLE_TEXT_BOX, null);
+    titleTextBox.addValidator(MandatoryValidator.class).addValidator(ECMNameValidator.class).addValidator(XSSValidator.class);
```

Comment:
Why did you add ECMNameValidator ?

Reply:
> Why did you add ECMNameValidator ?

The addition of the ECMNameValidator is mandatory so that the name of the document is compatible with jcr.

Reply:
In UINewDocumentForm, the title entered by the user is already updated to fit JCR constraint  :

>       title = getFileName(title, template);

So, no need to add a validator to check it.
The only needed validator is the one for XSS